pkg/asset/manifests/cloudproviderconfig.go

```diff
@@ -92,6 +92,17 @@ func (cpc *CloudProviderConfig) Generate(dependencies asset.Parents) error {
 		}
 
 		cm.Data[cloudProviderConfigDataKey] = openstackmanifests.CloudProviderConfig(cloud.CloudConfig)
+
+		// Get the ca-cert-bundle key if there is a value for cacert in clouds.yaml
+		caPath := cloud.CloudConfig.CACertFile
+		var caFile string
+		if caPath != "" {
+			caFile, err = openstackmanifests.GetCaCertFromDisk(caPath)
+			if err != nil {
+				return errors.Wrap(err, "failed to read clouds.yaml ca-cert from disk")
+			}
+		}
+		cm.Data["ca-bundle.pem"] = caFile
```

Comment:
If `cacert` is not present in `clouds.yaml`, you end up with an empty entry in `cm.Data`. What about moving this line into the conditional block instead?

```Go
// Get the ca-cert-bundle key if there is a value for cacert in clouds.yaml
if caPath := cloud.CloudConfig.CACertFile; caPath != "" {
	caFile, err := ioutil.ReadFile(caPath)
	if err != nil { ... }
	cm.Data["ca-bundle.pem"] = string(caFile)
}
```